Library/Homebrew/cli_parser.rb

```diff
@@ -58,7 +59,7 @@ def switch(*names, description: nil, env: nil, required_for: nil, depends_on: ni
           set_constraints(name, required_for: required_for, depends_on: depends_on)
         end
 
-        enable_switch(*names) if !env.nil? && !ENV["HOMEBREW_#{env.to_s.upcase}"].nil?
+        enable_switch(*names, source: :env_var) if !env.nil? && !ENV["HOMEBREW_#{env.to_s.upcase}"].nil?
```

Comment:
would prefer to extract these env/cli values into constants somewhere, but not sure what the style for that is, or even if a simple boolean would be better here

Reply:
I think this phrasing is clearer:

Would it might be better to extract `:env` and `:args` to constants, to avoid having magic values floating around, or even just to change it to a boolean `is_env_var?` flag?

Reply:
Sorry, still being stupid here! Can you give some examples of what values you think should maybe be constants?

Reply:
something like:

```
class Parser
  SWITCH_SOURCE_ENV_VAR = :env 
  SWITCH_SOURCE_CLI_ARG = :arg 

  ...

  def switch(...)
    ...
    enable_switch(*names, source: SWITCH_SOURCE_ENV_VAR)
  end
  
  ...

  def check_conflicts
    ...
    env_var_options = violations.select do |option|
      @switch_sources[option_to_name(option)] == SWITCH_SOURCE_ENV_VAR
    end
  end
end
```

Just to avoid having `:env` and `:arg` scattered around the code as magic values

Reply:
Feels a bit overkill. I think the symbols are fine.

Reply:
Sounds good. This is all set from my perspective, last thing would just be making sure that y'all are okay with the `brew style` failure, or if it would be better to update https://github.com/Homebrew/brew/pull/5738/files#diff-40e44d1232a302a388e451cc3a0a22f9R186 to `expect(...).to be_switch_a`, etc.

Reply:
@BenMusch Need to have `brew style` be passing I'm afraid, thanks. `brew style --fix` may be able to do that for you automatically.